timeseries/src/autogluon/timeseries/models/local/statsmodels.py

```diff
@@ -261,55 +287,49 @@ class ThetaModel(AbstractStatsmodelsModel):
         When set to -1, all CPU cores are used.
     """
 
-    statsmodels_allowed_init_args = [
+    allowed_local_model_args = [
         "deseasonalize",
         "seasonal_period",
         "use_test",
         "method",
         "difference",
     ]
-    statsmodels_allowed_fit_args = []
-
-    def _update_sm_model_init_args(self, sm_model_init_args: dict, data: TimeSeriesDataFrame) -> dict:
-        sm_model_init_args = sm_model_init_args.copy()
-        sm_model_init_args.setdefault("deseasonalize", True)
-
-        # Infer seasonal_period if seasonal_period is not given / is set to None
-        seasonal_period = sm_model_init_args.pop("seasonal_period", None)
-        if seasonal_period is None:
-            seasonal_period = get_seasonality(data.freq)
-        sm_model_init_args["period"] = seasonal_period
-
-        return sm_model_init_args
-
-    def _fit_local_model(
-        self, timeseries: pd.Series, sm_model_init_args: dict, sm_model_fit_args: dict
-    ) -> FittedLocalModel:
-        # ThetaModel in statsmodels doesn't provide a way to initialize the model with trained parameters,
-        # so we delegate training to `_predict_with_local_model`
-        if sm_model_init_args["deseasonalize"] and len(timeseries) < 2 * sm_model_init_args["period"]:
-            sm_model_init_args = sm_model_init_args.copy()
-            sm_model_init_args["deseasonalize"] = False
-        return FittedLocalModel(model_name=self.name, sm_model_init_args=sm_model_init_args, parameters=None)
 
+    def _update_local_model_args(
+        self, local_model_args: Dict[str, Any], data: TimeSeriesDataFrame, **kwargs
+    ) -> Dict[str, Any]:
+        local_model_args.setdefault("deseasonalize", True)
```

Comment:
have we decided dropping the copy operations is safe, or does the abstract model handle these?

Reply:
Yes, `_predict_with_local_model` always receives a copy of the `_local_model_args` dict. 

We're fine with modifying the dict inside `_update_local_model_args`, this only happens once and doesn't affect the `self._get_model_params()`.